### ISSUE TITLE
IIIF is broken

### ISSUE THREAD

**Issue by iangilman** [openseadragon/openseadragon#2065]
It looks like https://github.com/openseadragon/openseadragon/pull/2059 has broken IIIF (see https://github.com/openseadragon/openseadragon/issues/1985#issuecomment-975212887). We'll want to back out that change, but we should also see if we can solve the problem that PR was trying to fix in a different way.

**Comment by iangilman** [comment]
It looks like when I reviewed #2059, I misunderstood how `getLevelScale` is being used. I think we should return that function to its earlier state. In order to support the scenario that PR was intended to support, I propose we add a `setMaxLevel` function to `TileSource`, and it should trigger a fresh memoization for `getLevelScale`. What do you think, @kim-sanghoon ?

**Comment by kim-sanghoon** [comment]
@iangilman Thank you for letting me know! I should have tested the PR with other types of image slides...

Based on your proposal, I added a `setMaxLevel` function to `TileSource` as https://github.com/kim-sanghoon/openseadragon/commit/de39597c141cdf9e34c8a3b94ae18af4e2db048c, and invoked the function inside the constructor so that the `getLevelScale` memoization could be made at the initialization time.

I tested the code with the automated unit tests and demo pages, and it seems to be working. However, I personally feel my changes could be improved in terms of code quality. I will open a PR on this issue if the code is okay to be reviewed.
